src/EcsRx.Plugins.Persistence/Transformers/FromEntityDatabaseDataTransformer.cs

```diff
@@ -24,7 +24,15 @@ public object Transform(object converted)
             entityDatabaseData.EntityCollections
                 .Select(EntityCollectionDataTransformer.Transform)
                 .Cast<IEntityCollection>()
-                .ForEachRun(entityDatabase.AddCollection);
+                .ForEachRun(x =>
+                {
+                    if (entityDatabase.Collections.Any(e => e.Id == x.Id))
```

Comment:
This looks ok on first pass but im not entirely sure in the long run if we are safe to assume we want to delete the existing collection and replace, this being said for now this is great and will be accepted, but maybe in a future sweep it can be revisited to support potential additive/merge scenarios too.

Reply:
Yes, this temporary change is just to make this to work. Whats your plan for this to work if duplicate colletion is being loaded?

Reply:
I don't really have a plan to be honest, I initially assumed anyone using this would be going into a completely blank DB so there should never be any existing data, if I had to think of a solution it should probably just check if a collection with the id already exists, if so then just add each entity in which is missing (based on id), but thats assuming a lot and may not be correct for all use cases... this was meant to be an out the box example for just loading in state to an empty database i.e like Unity scenes, then you do whatever you want on top of that once its loaded.

Like with all use cases though its assumed the more advanced scenarios would probably just go write their own implementations for the same interfaces etc so im not too fussed right now about solving the problem as people can easily provide their own implementations but I am open to extending this default implementation further if it helps others, but as you say for right now this will get it working with existing data.